include/Basis/SpinorBasis/SpinComponent.hpp

```diff
@@ -0,0 +1,32 @@
+// This file is part of GQCG-gqcp.
+// 
+// Copyright (C) 2017-2019  the GQCG developers
+// 
+// GQCG-gqcp is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// GQCG-gqcp is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with GQCG-gqcp.  If not, see <http://www.gnu.org/licenses/>.
+// 
+#pragma once
+
+
+namespace GQCP {
+
+/**
+ *  An enumeration containing the modes of spin for electrons
+ */
+enum SpinComponent {
+    ALPHA = 0,
+    BETA = 1
+};
```

Comment:
`enum class` does not have automatic conversion to `int`and would require static casts.

Reply:
Oh, I see. Could you check the code that is using `CartesianDirection` and see if a similar reasoning could be applied to also change it from `enum class` to `enum`? If it's too much work, feel free to open an issue addressing it.

Reply:
In `CartesianExponents` you have 3 individual members x,y,z instead of an `array<size_t, 3>` and you chose via `switch` do you suggest we use an array and accessed via `enum` or do you suggest `SpinComponent` as `enum class` and using switch? I personally am a fan of an array and accessed via `enum`.

Reply:
@tmhuysen I also favor the `array` approach as well.